File: finafanBE/src/main/java/com/a702/finafanbe/core/funding/funding/dto/EntertainerResponse.java
package com.a702.finafanbe.core.funding.funding.dto;

public record EntertainerResponse (
        Long entertainerId,
        String name,
        String imageUrl,
        String thumbnailUrl
) {
    public static EntertainerResponse of (
            Long entertainerId,
            String name,
            String imageUrl,
            String thumbnailUrl
    ) {
        return new EntertainerResponse(
                entertainerId,
                name,
                imageUrl,
                thumbnailUrl
        );
    }
}

File: finafanAndroid/app/src/main/java/com/a702/finafan/data/funding/dto/response/EntertainerResponse.kt
package com.a702.finafan.data.funding.dto.response

import com.a702.finafan.domain.funding.model.Star

data class EntertainerResponse (
    val entertainerId: Long,
    val name: String,
    val imageUrl: String,
    val thumbnailUrl: String
)

fun EntertainerResponse.toDomain() : Star {
    return Star(
        id = this.entertainerId,
        name = this.name,
        index = 0,
        image = this.imageUrl,
        thumbnail = this.thumbnailUrl
    )
}
